### ISSUE TITLE
Is router service broken?

### ISSUE THREAD

**Issue by KalachevDev** [emberjs/ember.js#18177]
**TL;DR:**

> Router service does double transition when navigate to route with queryParameter ` queryParam: { refreshModel: true } `and throws and error after abort of the first transition. 

==================
A lot of text below :)

Let's say I have some link  component that triggers transition to /page-with-refresh-model-query-param?query=123 and in page-with-refresh-model-query-param/route.js I have query: { refreshModel: true }. Transition works fine in general, but there is one issue: in link component I have following code: this.router.transitionTo(...).catch(err => console.log(err)); in catch block here I receive an error with message: TransitionAborted: TransitionAborted. After some digging in source code I was able to figure out an issue but It seems strange to me. First of all router triggers transition to page-with-refresh-model-query-param after that it triggers queryParamsDidChange that defined in Ember itself and Ember refreshes router there because some queryParams were changed (from default value to 123 in our case) and triggers second transition for the same destination. As a result first transition is aborted and I receive an error in catch chain in link component (but it's not a real error just a transition abortion)
Should I rely on error name: TransitionAborted and handle it different way than plain error?
But it still seems odd to me that I get this "error" when there is no real error or abort

[Twiddle](https://ember-twiddle.com/69c2db40bfe7a292916ba7416c489435?openFiles=templates.index.hbs%2Ctemplates.components.my-link.hbs)

**Comment by NullVoxPopuli** [comment]
After playing with the twiddle, this seems like a false negative.

**Comment by KalachevDev** [comment]
`this.transitionTo` works the same in routes.

**Comment by linearza** [comment]
Related issue: https://github.com/emberjs/ember.js/issues/15801

**Comment by rwjblue** [comment]
Thank you @linearza, closing this as a duplicate of #15801
